packages/fluentui/react-northstar/src/components/Menu/Menu.tsx

```diff
@@ -90,160 +105,203 @@ export interface MenuProps extends UIComponentProps, ChildrenComponentProps {
+    className: menuClassName,
+    mapPropsToStyles: () => ({
+      iconOnly,
+      fluid,
+      pointing,
+      pills,
+      primary,
+      underlined,
+      vertical,
+      submenu,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-  static Item = MenuItem;
-  static Divider = MenuDivider;
+  const [activeIndex, setIndex] = useAutoControlled({
+    defaultValue: props.defaultActiveIndex,
+    value: props.activeIndex,
+    initialValue: undefined,
+  });
 
-  setActiveIndex = (e: React.SyntheticEvent, activeIndex: number) => {
-    _.invoke(this.props, 'onActiveIndexChange', e, { ...this.props, activeIndex });
-    this.setState({ activeIndex });
+  const setActiveIndex = (e: React.SyntheticEvent, activeIndex: number) => {
+    _.invoke(props, 'onActiveIndexChange', e, { ...props, activeIndex });
+    setIndex(activeIndex);
   };
 
-  handleItemOverrides = variables => predefinedProps => ({
+  const handleItemOverrides = variables => predefinedProps => ({
     onClick: (e, itemProps) => {
       const { index } = itemProps;
 
-      this.setActiveIndex(e, index);
+      setActiveIndex(e, index);
 
-      _.invoke(this.props, 'onItemClick', e, itemProps);
+      _.invoke(props, 'onItemClick', e, itemProps);
       _.invoke(predefinedProps, 'onClick', e, itemProps);
     },
     onActiveChanged: (e, props) => {
       const { index, active } = props;
       if (active) {
-        this.setActiveIndex(e, index);
-      } else if (this.state.activeIndex === index) {
-        this.setActiveIndex(e, null);
+        setActiveIndex(e, index);
+      } else if (activeIndex === index) {
+        setActiveIndex(e, null);
       }
       _.invoke(predefinedProps, 'onActiveChanged', e, props);
     },
     variables: mergeComponentVariables(variables, predefinedProps.variables),
   });
 
-  handleDividerOverrides = variables => predefinedProps => ({
+  const handleDividerOverrides = variables => predefinedProps => ({
     variables: mergeComponentVariables(variables, predefinedProps.variables),
   });
 
-  renderItems = (
-    styles: ComponentSlotStylesPrepared,
-    variables: ComponentVariablesObject,
-    accessibility: ReactAccessibilityBehavior,
-  ) => {
-    const {
-      iconOnly,
-      items,
-      pills,
-      pointing,
-      primary,
-      secondary,
-      underlined,
-      vertical,
-      submenu,
-      indicator,
-    } = this.props;
-    const { activeIndex } = this.state;
+  const renderItems = () => {
     const itemsCount = _.filter(items, item => getKindProp(item, 'item') !== 'divider').length;
     let itemPosition = 0;
 
-    const overrideItemProps = this.handleItemOverrides(variables);
-    const overrideDividerProps = this.handleDividerOverrides(variables);
+    const overrideItemProps = handleItemOverrides(variables);
+    const overrideDividerProps = handleDividerOverrides(variables);
```

Comment:
```suggestion
```

Reply:
You have regressions there because previously `variables` were `resolvedVariables`, now it's `props.variables`. There is no way to get `resolvedVariables` from `useStyles` and should not be.

To solve it please `menuItemVariables.ts` with reexport to Teams Dark theme. And ensure that `menuDividerVariables.ts` exists in all three themes.

Reply:
Thanks for the explanation. 

@pompomon FYI

Reply:
Before I solve it in menuItems... Let's merge it as functional and I do it in the latest version:

https://github.com/microsoft/fluentui/pull/13070